junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/LauncherDiscoveryRequestBuilder.java

```diff
@@ -204,15 +205,23 @@ else if (filter instanceof DiscoveryFilter<?>) {
 	public LauncherDiscoveryRequest build() {
 		LauncherConfigurationParameters launcherConfigurationParameters = new LauncherConfigurationParameters(
 			this.configurationParameters);
+		LauncherDiscoveryListener discoveryListener = getLauncherDiscoveryListener(launcherConfigurationParameters);
 		return new DefaultDiscoveryRequest(this.selectors, this.engineFilters, this.discoveryFilters,
-			this.postDiscoveryFilters, launcherConfigurationParameters, getLauncherDiscoveryListener());
+			this.postDiscoveryFilters, launcherConfigurationParameters, discoveryListener);
 	}
 
-	private LauncherDiscoveryListener getLauncherDiscoveryListener() {
-		if (this.discoveryListeners.isEmpty()) {
-			return LauncherDiscoveryListeners.logging();
+	private LauncherDiscoveryListener getLauncherDiscoveryListener(ConfigurationParameters configurationParameters) {
+		LauncherDiscoveryListener defaultDiscoveryListener = LauncherDiscoveryListeners.fromConfigurationParameter(
```

Comment:
Not sure about the naming - would `configuredDiscoveryListener ` or `configuredDefaultDiscoveryListener ` convey the intent more clearly?

Reply:
TBH I think it's clear enough. After all it mimics the new configuration parameter.